File: android-nosql/src/main/java/com/github/florent37/androidnosql/datasaver/DataSaver.java
package com.github.florent37.androidnosql.datasaver;

import java.util.Set;

/**
 * Created by florentchampigny on 24/05/2017.
 */

public interface DataSaver {
    void saveNodes(String completePath, Set<String> value);

    void saveValue(String completePath, Object value);

    Set<String> getNodes();

    Object getValue(String completePath);

    void remove(String startingPath);
}

File: android-nosql/src/main/java/com/github/florent37/androidnosql/datasaver/DataSaver.kt
package com.github.florent37.androidnosql.datasaver

interface DataSaver {
    suspend fun saveNodes(completePath: String, value: Set<String>?)
    suspend fun saveValue(completePath: String, value: Any?)
    val nodes: Set<String>
    suspend fun getValue(completePath: String): Any?
    suspend fun remove(startingPath: String)
}